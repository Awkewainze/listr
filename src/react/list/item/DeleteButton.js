import React from 'react';
import './DeleteButton.scss';

const DeleteButton = ({ onDelete, itemId }) => {
  const onDeleteClick = () => {
    onDelete(itemId);
  };

  return (
    <button className="DeleteButton" type="button" onClick={onDeleteClick}>
      Delete
    </button>
  );
};

export default DeleteButton;