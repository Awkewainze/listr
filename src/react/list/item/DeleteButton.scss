.DeleteButton {
  margin: 4px 12px;
  display: flex;
  align-content: center;
  justify-content: center;

  .DeleteButton__inner {
    outline: none;
    border: none;
    background-color: #e24141;
    border-radius: 6px;
    font-size: 12px;
    color: #fff;
    cursor: pointer;
    padding: 8px;
  }
}