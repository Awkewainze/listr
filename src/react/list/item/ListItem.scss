@import '../../../variables';
@import '../../../mixins';

.ListItem {
  display: flex;
  border-bottom: 1px solid #afafaf;
  margin: 6px 0;
  justify-content: flex-start;
  padding: 0 20px;

  @include minWidth($size-mobile) {
    .DeleteButton {
      opacity: 0;
    }

    &:hover {
      border-color: #7f7f7f;

      .DeleteButton {
        opacity: 1;
      }
    }
  }

  .ListItem__text {
    padding: 20px 0;
  }

  .ListItem__checkbox {
    cursor: pointer;
    margin-right: 12px;
  }

  .ListItem__text {
    flex: 1;
    overflow: scroll;
    margin: 12px 0;
    line-height: 26px;
  }

  .ListItem__text--checked {
    text-decoration: line-through;
    opacity: 0.75;
  }

  @include minWidth($size-mobile) {
    .ListItem__checkbox {
      min-width: 72px;
      margin-right: 0;
    }
  }
}