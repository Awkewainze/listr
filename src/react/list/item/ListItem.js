import React from 'react';
import './ListItem.scss';
import Checkbox from './Checkbox';
import DeleteButton from './DeleteButton';

const ListItem = ({
  item,
  itemId,
  index,
  checked = false,
  onCheck,
  onDelete
}) => (
  <div className="ListItem">
    <Checkbox className="ListItem__checkbox" checked={checked} index={index} itemId={itemId} onChange={onCheck} />
    <DeleteButton itemId={itemId} onDelete={onDelete} />
    <p className={`ListItem__text ${checked ? 'ListItem__text--checked' : ''}`}>
      {item}
    </p>
  </div>
);

export default ListItem;