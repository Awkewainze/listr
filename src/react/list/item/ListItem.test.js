import React from 'react';
import { shallow } from 'enzyme';
import ListItem from './ListItem';
import DeleteButton from './DeleteButton';

const onCheck = jest.fn();

const TEST_PROPS = {
  onCheck: jest.fn(),
  onDelete: jest.fn(),
  index: 1,
  itemId: '123',
  checked: false,
  draggableProps: {},
  dragHandleProps: {},
  ref: null
};

it('matches snapshot', () => {
  expect(shallow(<ListItem {...TEST_PROPS} />)).toMatchSnapshot();
});

it('is unchecked correctly', () => {
  const container = shallow(<ListItem {...TEST_PROPS} />);
  expect(container.find('.ListItem__checkbox').prop('checked')).toBeFalsy();
  expect(container.find('p').hasClass('ListItem__text--checked')).toBeFalsy();
});

it('is checked correctly', () => {
  const container = shallow(<ListItem {...TEST_PROPS} checked={true} />);
  expect(container.find('.ListItem__checkbox').prop('checked')).toBeTruthy();
  expect(container.find('p').hasClass('ListItem__text--checked')).toBeTruthy();
});