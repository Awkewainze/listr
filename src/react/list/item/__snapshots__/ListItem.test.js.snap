// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="ListItem"
>
  <div
    className="ListItem__draggable"
  >
    :
  </div>
  <Checkbox
    checked={false}
    className="ListItem__checkbox"
    index={1}
    itemId="123"
    onChange={[MockFunction]}
  />
  <p
    className="ListItem__text "
  >
    oranges
  </p>
  <DeleteButton
    itemId="123"
  />
</div>
`;
